pallets/dkg-metadata/src/lib.rs

```diff
@@ -351,7 +351,11 @@ impl<T: Config> Pallet<T> {
 		// As in GRANDPA, we trigger a validator set change only if the the validator
 		// set has actually changed.
 
-		if new != Self::authorities() {
+		let changed = new != Self::authorities();
+		#[cfg(test)]
```

Comment:
Here too, this should be avoided. It could potentially cause deviation in runtime binaries.